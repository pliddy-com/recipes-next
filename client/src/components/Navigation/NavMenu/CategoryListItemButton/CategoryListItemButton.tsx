import { MouseEventHandler } from 'react';

import Link from 'next/link';

import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { Maybe } from 'types/queries';

interface CategoryListItemButtonProps {
  onClick: MouseEventHandler;
  root: Maybe<string> | undefined;
  slug: Maybe<string> | undefined;
  title: Maybe<string> | undefined;
  total?: number;
}

const CategoryListItemButton = ({
  onClick,
  root,
  slug,
  title,
  total
}: CategoryListItemButtonProps) =>
  slug && title && onClick ? (
    <ListItemButton
      aria-label={`${slug} category`}
      component={Link}
      href={`/${root}/${slug}`}
      onClick={onClick}
    >
      <ListItemText primary={`${title} (${total})`} />
    </ListItemButton>
  ) : null;

export default CategoryListItemButton;
