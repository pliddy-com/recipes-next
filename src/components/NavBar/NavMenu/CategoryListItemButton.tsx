import Link from 'next/link';

import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { Maybe } from 'types/generated/graphql';

type CategoryListItemButtonProps = {
  onClick: VoidFunction;
  slug: Maybe<string> | undefined;
  title: Maybe<string> | undefined;
  total?: number;
};

const CategoryListItemButton = ({
  onClick,
  slug,
  title,
  total,
}: CategoryListItemButtonProps) => {
  return (
    <ListItemButton
      component={Link}
      href={`/category/${slug}`}
      onClick={onClick}
    >
      <ListItemText primary={`${title} (${total})`} />
    </ListItemButton>
  );
};

export default CategoryListItemButton;
