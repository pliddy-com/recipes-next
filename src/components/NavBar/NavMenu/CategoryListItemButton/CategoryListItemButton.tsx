import Link from 'next/link';

import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { Maybe } from 'types/generated/graphql';

interface CategoryListItemButtonProps {
  onClick: VoidFunction;
  slug: Maybe<string> | undefined;
  title: Maybe<string> | undefined;
  total?: number;
}

const CategoryListItemButton = ({
  onClick,
  slug,
  title,
  total,
}: CategoryListItemButtonProps) => {
  return (
    <ListItemButton
      component={Link}
      href={`/category/${slug}`}
      onClick={onClick}
    >
      <ListItemText primary={`${title} (${total})`} className="menuItem" />
    </ListItemButton>
  );
};

export default CategoryListItemButton;
