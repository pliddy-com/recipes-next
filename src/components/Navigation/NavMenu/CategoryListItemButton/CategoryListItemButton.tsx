import Link from 'next/link';

import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { Maybe } from 'types/queries';

interface CategoryListItemButtonProps {
  onClick: VoidFunction;
  root: Maybe<string> | undefined;
  slug: Maybe<string> | undefined;
  title: Maybe<string> | undefined;
  total?: number;
}

const CategoryListItemButton = ({
  onClick,
  root,
  slug,
  title,
  total,
}: CategoryListItemButtonProps) =>
  slug && title && onClick ? (
    <ListItemButton
      aria-label={`${slug} category`}
      component={Link}
      href={`/${root}/${slug}`}
      onClick={onClick}
    >
      <ListItemText primary={`${title} (${total})`} className="menuItem" />
    </ListItemButton>
  ) : null;

export default CategoryListItemButton;
