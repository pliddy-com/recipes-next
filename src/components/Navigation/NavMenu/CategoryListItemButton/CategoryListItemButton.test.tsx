// import testing-library methods
import { fireEvent, render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import CategoryListItemButton from './CategoryListItemButton';

describe('CategoryListItemButton', () => {
  const callback = jest.fn();

  describe('when there is a correct payload', () => {
    it('it renders a list item for the menu', () => {
      const slug = 'slug';
      const title = 'Test Title';
      const numRecipes = 10;
      const root = 'category';

      const expectedHref = `/${root}/${slug}`;

      const { container, queryByRole } = render(
        <CategoryListItemButton
          slug={slug}
          title={title}
          onClick={callback}
          total={numRecipes}
          root={root}
        />
      );

      // assert that component has been rendered
      const link = queryByRole('link', { name: `${slug} category` });
      expect(link).toBeInTheDocument();

      // assert that component has correct href
      expect(link).toHaveAttribute('href', expectedHref);

      // assert that the callback is called on click
      link && fireEvent.click(link);
      expect(callback).toBeCalled();

      // assert that the component matches the existing snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe('when there is not a correct payload', () => {
    it('it does not render', () => {
      const slug = null;
      const root = null;

      const { queryByRole } = render(
        <CategoryListItemButton
          root={root}
          slug={slug}
          title={null}
          onClick={callback}
        />
      );

      // assert that the component is not rendered
      const link = queryByRole('link', { name: `${slug} category` });
      expect(link).toBeNull();
    });
  });
});
