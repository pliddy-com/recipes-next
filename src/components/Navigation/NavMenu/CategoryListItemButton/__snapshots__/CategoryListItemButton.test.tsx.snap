// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItemButton when there is a correct payload it renders a list item for the menu 1`] = `
<div>
  <a
    aria-label="slug category"
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
    href="/tag/slug"
    tabindex="0"
  >
    <div
      class="MuiListItemText-root menuItem css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Test Title (10)
      </span>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</div>
`;
